pkg/build/api/v1/types.go

```diff
@@ -371,6 +375,13 @@ type SourceBuildStrategy struct {
 	ForcePull bool `json:"forcePull,omitempty" description:"forces the source build to pull the image if true"`
 }
 
+// BuildPostCommitSpec holds a build post commit hook specification.
+type BuildPostCommitSpec struct {
+	// RunBash holds a bash script to be executed in a container running the
+	// build output image.
+	RunBash string `json:"runBash,omitempty" description:"bash script to be executed in a container running the build output image"`
```

Comment:
I think we want to treat /bin/sh or /bin/bash as the explicit entrypoint, with the following behaviors:

* `command: ["/foo"]` - runs `/foo` in the container, no shell
* `args: ["bundle", "exec", "rake", "test"]` runs `/bin/bash -c bundle exec rake test` in the current working directory - this is "here's the shell command line I want you to run"
* specifying both command and args appends args to command and runs them

That means most users specify args and get exactly what they want.  Some users can specify command and get exactly what they want.  A few users *may* want to allow command to omit the leading prefix and be relative to the working directory - I think we should allow this as well.